### PR TITLE
Add AppendInt function with positive/negative number convertion to string

### PR DESCRIPTION
**Pull request by ReneWerner87**
```md
Benchmark_ItoA/fiber             (pos_num)-12  186969812        6.25 ns/op       0 B/op       0 allocs/op
Benchmark_ItoA/fiber             (pos_num)-12  193965686        6.16 ns/op       0 B/op       0 allocs/op
Benchmark_ItoA/strconv.Itoa      (pos_num)-12   80716807       14.42 ns/op       4 B/op       1 allocs/op
Benchmark_ItoA/strconv.Itoa      (pos_num)-12   80445802       14.85 ns/op       4 B/op       1 allocs/op
Benchmark_ItoA/strconv.FormatInt_(pos_num)-12   81137728       14.52 ns/op       4 B/op       1 allocs/op
Benchmark_ItoA/strconv.FormatInt_(pos_num)-12   81345360       14.51 ns/op       4 B/op       1 allocs/op
Benchmark_ItoA/fiber             (neg_num)-12  192902048        6.18 ns/op       0 B/op       0 allocs/op
Benchmark_ItoA/fiber             (neg_num)-12  194245189        6.14 ns/op       0 B/op       0 allocs/op
Benchmark_ItoA/strconv.Itoa      (neg_num)-12   84505304       13.55 ns/op       5 B/op       1 allocs/op
Benchmark_ItoA/strconv.Itoa      (neg_num)-12   82524801       13.54 ns/op       5 B/op       1 allocs/op
Benchmark_ItoA/strconv.FormatInt (neg_num)-12   84884136       13.63 ns/op       5 B/op       1 allocs/op
Benchmark_ItoA/strconv.FormatInt (neg_num)-12   85829492       13.63 ns/op       5 B/op       1 allocs/op
```

## Summary by CodeRabbit

- **New Features**
  - Introduced a function to append integer string representations to byte slices, supporting both positive and negative integers.
  - Added comprehensive benchmark tests for integer-to-string conversion, enhancing performance analysis capabilities.

- **Bug Fixes**
  - Minor formatting adjustment in comment lines for improved clarity.

- **Tests**
  - Added new test cases to validate the functionality of the new integer appending feature.
  - Introduced benchmark tests comparing the performance of the new function against standard library methods.

### CODE REVIEW

File: convert.go [hunk 1]
```diff
@@ -151,3 +151,34 @@ func ToString(arg any, timeFormat ...string) string {
 		return fmt.Sprint(arg)
 	}
 }
+
+// AppendInt appends the string representation of the int n to dst and returns the extended buffer.
+func AppendInt(dst []byte, n int) []byte {
```
Comment (gaby) [inline convert.go:156]: I think we need a separate function for `AppendUint` that uses `uint` as param, else we won't be able to handle values bigger than `2,147,483,647`

File: convert_test.go [hunk 2]
```diff
@@ -295,3 +309,70 @@ func Benchmark_UnsafeString(b *testing.B) {
...
+		b.ReportAllocs()
+		b.ResetTimer()
+
+		for n := 0; n < b.N; n++ {
+			resB = AppendInt(resB[:0], number)
+		}
+		require.Equal(b, []byte(numberString), resB)
+	})
+
+	b.Run("default - strconv.Itoa (positiv number)", func(b *testing.B) {
```
Comment (gaby) [inline convert_test.go:334]: positive*

File: convert_test.go [hunk 3]
```diff
@@ -295,3 +309,70 @@ func Benchmark_UnsafeString(b *testing.B) {
...
+		b.ReportAllocs()
+		b.ResetTimer()
+		for n := 0; n < b.N; n++ {
+			resS = strconv.Itoa(number)
+		}
+		require.Equal(b, numberString, resS)
+	})
+
+	b.Run("default - strconv.FormatInt (positiv number)", func(b *testing.B) {
```
Comment (gaby) [inline convert_test.go:343]: positive*

File: convert.go (hunk 1 above)
Comment (gaby) [reply convert.go:156]: The `math` module in Golang defines these:

```go
MaxUint32 = 1<<32 - 1           // 4294967295
MaxUint64 = 1<<64 - 1           // 18446744073709551615
```

**Comment by ReneWerner87** [comment]
there is already a AppendInt function in strconv
https://pkg.go.dev/strconv#AppendInt

4 digit number
image

2 digit number
image

**Pull request closed without merging**